### ISSUE TITLE
Graphical Glitch for spear weilding goblin

### ISSUE THREAD

**Issue by Monkooky** [crawl/crawl#2557]
Webtiles
Version: Star experimental
![image](https://user-images.githubusercontent.com/6244827/170612995-dc894050-3e8f-4ded-ba3e-440b70b079e0.png)

This goblin's spear was missing the butt one frame, then had it back the next. (Unfortunately killed it the hit after noticing, so don't have screenshots)
![image](https://user-images.githubusercontent.com/6244827/170612849-1afa6eae-8fad-4105-b707-387f4a5115ff.png)

Once the goblin is slain, the spearbutt remains.
![image](https://user-images.githubusercontent.com/6244827/170612888-31bd47b7-2c7c-4a5c-b6ff-2bbba27d3fe0.png)

**Comment by tr2424** [comment]
I have noticed this with spear-wielding goblins as well. Sometimes when the move around, the butt of the spear, i.e. basically a pixel gets left behind and can leave a trail.

Oddly enough, I just noticed this same issue with a named mob "Donald"... he had a scimitar or falchion I believe that left a small gold dot (pixel) behind while he moved. That mob is not a goblin though (least I don't think it is) I did not slay him so don't know if the same behavior would occur. No screenshots sorry, but figured I'd let someone know.
